source/Locales/LawsonLocale/CompactElementsOfPoint.lagda

```diff
@@ -0,0 +1,425 @@
+
+\begin{code}
+
+ Point : 𝓤 ⁺  ̇
+ Point = 𝟏L ─c→  X
+
+\end{code}
+
+We now proceed to the definition of the family mentioned in the preamble. We
+work with a dcpo `𝓓` that is assumed to
+
+  - have a bottom element,
+  - be a Scott domain, and
+  - satisfy the aforementioned decidability condition for upper boundedness.
+
+\begin{code}
+
+open DefinitionOfScottDomain
+
+module Construction
+        (𝓓    : DCPO {𝓤 ⁺} {𝓤})
+        (ua   : Univalence)
+        (hl   : has-least (underlying-order 𝓓))
+        (sd   : is-scott-domain 𝓓 holds)
+        (dc   : decidability-condition 𝓓) where
+
+ open SpectralScottLocaleConstruction₂ 𝓓 ua hl sd dc pe
+ open SpectralScottLocaleConstruction 𝓓 hl hscb dc bc pe hiding (scb; β; ϟ)
+ open DefnOfScottTopology 𝓓 𝓤
+
+ open Preliminaries σ⦅𝓓⦆
+ open Properties 𝓓
+
+\end{code}
+
+We denote by `B𝓓` the basis of `𝓓`.
+
+\begin{code}
+
+ B𝓓 : Fam 𝓤 ⟨ 𝓓 ⟩∙
+ B𝓓 = index-of-compact-basis 𝓓 hscb , family-of-compact-elements 𝓓 hscb
+
+\end{code}
+
+We use the abbreviation `scb` for the proof that `B𝓓` is a small basis
+consisting of compact opens.
+
+\begin{code}
+
+ scb : is-small-compact-basis 𝓓 (family-of-compact-elements 𝓓 hscb)
+ scb = small-compact-basis 𝓓 hscb
+
+\end{code}
+
+By `βₖ i`, we denote the element denoted by an index `i`, packaged up with the
+proof that it is compact.
+
+\begin{code}
+
+ open is-small-compact-basis scb
+ open BottomLemma 𝓓 𝕒 hl
+
+ βₖ : (i : index B𝓓) → Σ c ꞉ ⟨ 𝓓 ⟩∙ , is-compact 𝓓 c
+ βₖ i = B𝓓 [ i ] , basis-is-compact i
+
+\end{code}
+
+We now write down the family of compact opens of a point which we denote
+`𝒦-in-point`.
+
+\begin{code}
+
+ 𝒦-in-point : Point → Fam 𝓤 ⟨ 𝓓 ⟩∙
+ 𝒦-in-point (F , _) =
+  ⁅ B𝓓 [ i ] ∣ (i , _) ∶ (Σ i ꞉ index B𝓓 , ↑ˢ[ βₖ i ] ∈ₚ F holds) ⁆
+
+\end{code}
+
+The family `𝒦-in-point` is always inhabited.
+
+\begin{code}
+
+ open ScottLocaleProperties 𝓓 hl hscb pe
+
+ 𝒦-in-point-is-inhabited
+  : (ℱ@(F , _) : Point)
+  → is-inhabited (underlying-order 𝓓) (index (𝒦-in-point ℱ))
+ 𝒦-in-point-is-inhabited ℱ@(F , _) = ∥∥-rec ∃-is-prop † γ
+  where
+   Ⅲ : F 𝟏[ 𝒪 σ⦅𝓓⦆ ] ＝ ⊤
+   Ⅲ = frame-homomorphisms-preserve-top (𝒪 σ⦅𝓓⦆) (𝒪 𝟏L) ℱ
+
+   ζ : 𝟏[ 𝒪 σ⦅𝓓⦆ ] ∈ F
+   ζ = equal-⊤-gives-holds (F 𝟏[ 𝒪 σ⦅𝓓⦆ ]) Ⅲ
+
+   † : Σ i ꞉ index B𝓓 , B𝓓 [ i ] ＝ ⊥ᴰ → ∃ i ꞉ index B𝓓 , ↑ˢ[ βₖ i ] ∈ₚ F holds
+   † (i , p) = ∣ i , equal-⊤-gives-holds (F ↑ˢ[ βₖ i ]) ※ ∣
+    where
+     Ⅰ = ap
+          F
+          (to-subtype-＝
+            (holds-is-prop ∘ is-scott-open)
+            (ap (principal-filter 𝓓) p))
+     Ⅱ = ap F ↑⊥-is-top
+
+     ※ : F ↑ˢ[ βₖ i ] ＝ ⊤
+     ※ = F ↑ˢ[ βₖ i ]    ＝⟨ Ⅰ ⟩
+         F ↑ˢ[ ⊥ᴰ , ⊥κ ] ＝⟨ Ⅱ ⟩
+         F 𝟏[ 𝒪 σ⦅𝓓⦆ ]   ＝⟨ Ⅲ ⟩
+         ⊤               ∎
+
+   γ : ∃ i ꞉ index B𝓓 , B𝓓 [ i ] ＝ ⊥ᴰ
+   γ = small-compact-basis-contains-all-compact-elements 𝓓 (B𝓓 [_]) scb ⊥ᴰ ⊥κ
+
+\end{code}
+
+The family `𝒦-in-point` is closed under binary upper bounds.
+
+\begin{code}
+
+ closed-under-binary-upperbounds
```

Comment:
To make sure I understand you correctly, are you saying I should explain how `is-semidirected` is defined for the reader who might not be familiar with it? That might not be a bad idea since the term semidirected is not entirely standard, which is why I inadvertently spelled out what it says.

Reply:
No, sorry, I was a bit unclear. I think the situation is as follows: semidirectedness just says that the family contains _some_ upper bound of the pair; the name of the lemma seems to say that the family contains _all_ upper bounds of the pair; while the proof actually shows that the family contains the _least_ upper bound of the pair. Is that correct?
If all you care about is semidirectedness, then I would name the lemma something with `is-semidirected` as well.

Reply:
(Also, it should be `upper-bound` in the name.)

Reply:
> If all you care about is semidirectedness, then I would name the lemma something with is-semidirected as well.

That is indeed a subtle but important difference which I glossed over. I can see this being misleading. Thanks for your attention @tomdjong.

Reply:
Renamed to `𝒦-in-point-is-semidirected`.